src/include/daos/debug.h

```diff
@@ -10,7 +10,9 @@
 #include <stdio.h>
 #include <daos_errno.h>
 
+#ifndef DD_FAC
 #define DD_FAC(name)	daos_##name##_logfac
+#endif /* !DD_FAC */
```

Comment:
why is this needed now?

Reply:
It looks like there's a conflict between subsystems in the logging framework, e.g., macros for `st` (self_test) and  `mgmt` ?  In other words, the `ifndef` is a workaround for this error:

```
+ gcc -o build/dev/gcc/src/utils/self_test/self_test.o ... src/utils/self_test/self_test.c
In file included from src/include/daos/common.h:28:0,
                 from src/include/daos/drpc.h:11,
                 from src/include/daos/agent.h:10,
                 from src/utils/self_test/self_test.c:19:
src/include/daos/debug.h:13:0: error: "DD_FAC" redefined [-Werror]
 #define DD_FAC(name) daos_##name##_logfac
 ^
In file included from src/cart/crt_internal.h:14:0,
                 from src/cart/tests_common.h:18,
                 from src/utils/self_test/self_test.c:16:
src/cart/crt_debug.h:11:0: note: this is the location of the previous definition
 #define DD_FAC(name) crt_##name##_logfac
 ^
```

Reply:
we should probably unify all the debug stuff between cart/gurt/daos